### PR TITLE
feat(interlinks): filter on py domain by default

### PR DESCRIPTION
**Pull request by jmbuhr**
Types such as `dict` and `list` in the python standard library appear multiple times in the inventory. For example, for `dict`, there is the standard library https://docs.python.org/3.10/library/stdtypes.html#dict (the one we want), but also more obscure matches like https://docs.python.org/3.10/library/2to3.html?highlight=dict#to3fixer-dict.

Luckily, the `stdtypes` matches come up first. There is no guaranty for this, but I think it is fine to still link the first match and log a warning instead of not linking them.

While I was at it, I also made the module local functions local, renamed the `inventory` variable in the for loop to not shadow the global `inventory` and replaced deprecated `quarto.utils` functions with the appropriate `quarto.log` functions (had to get rid of those squiggly lines).

### CODE REVIEW

File: _extensions/interlinks/interlinks.lua [hunk 1]
```diff
@@ -31,7 +31,9 @@ function lookup(search_object)
             if search_object.domain and item.domain ~= search_object.domain then
                 goto continue
             else
-                table.insert(results, item)
+                if item.domain == "py" then
```
Comment (machow) [inline _extensions/interlinks/interlinks.lua:34]: Could we have this also check `search_object.domain or item.domain == "py"`? That way, people will be able to use the full link syntax still to link to non-python docs (as in [these examples](https://machow.github.io/quartodoc/get-started/interlinks.html#example-python.org-print))

**Review by machow — CHANGES_REQUESTED** [review]
This looks great, thanks! I left a very small comment, that I think would let us keep using non-python links. If that tweak works, we should be ready to merge!

File: _extensions/interlinks/interlinks.lua (hunk 1 above)
Comment (jmbuhr) [reply _extensions/interlinks/interlinks.lua]: Good point, fixed it.

**Comment by machow** [comment]
Thanks again for submitting! In case you're interested in trying it out, I have a PR open to speed up the interlinks filter, by having it just read raw inventory files :) https://github.com/machow/quartodoc/pull/253.

**Pull request merged**